src/cr/cube/measures/pairwise_significance.py

```diff
@@ -126,28 +129,83 @@ def summary_pairwise_indices(self):
-        col_margin_props = self._unweighted_col_margin / total_margin
+        col_margin_props = self._slice.column_base / self._slice.table_margin
         diff = col_margin_props - col_margin_props[self._col_idx]
-        var_props = col_margin_props * (1.0 - col_margin_props) / total_margin
+        var_props = (
+            col_margin_props * (1.0 - col_margin_props) / self._slice.table_margin
+        )
         se_diff = np.sqrt(var_props + var_props[self._col_idx])
         return diff / se_diff
 
     @lazyproperty
-    def summary_p_vals(self):
-        return 2 * (1 - t.cdf(abs(self.summary_t_stats), df=self._df))
+    def t_stats(self):
+        if self._slice.cube_is_mr_by_itself:
+            return self.t_stats_correct
+        props = self._slice.column_proportions
+        diff = props - props[:, [self._col_idx]]
+        var_props = props * (1.0 - props) / self._slice.column_base
+        se_diff = np.sqrt(var_props + var_props[:, [self._col_idx]])
+        return diff / se_diff
+
+    @lazyproperty
+    def t_stats_correct(self):
+        """It returns the t_statistic for MR variables considering the overlaps
+        """
+        diff, se_diff = self._slice.overlaps_tstats
+        t_stats = diff[:, self._col_idx, :] / se_diff[:, self._col_idx, :]
+        t_stats[:, self._col_idx] = 0
+        return t_stats
+
+    @lazyproperty
+    def t_stats_scale_means(self):
+        """
+        This property calculates the Two-tailed t-test using the formula:
+        t = X1 - X2 / Sx1x2 * sqrt(1/n1 + 1/n2)
+        where X1 and X2 are the scale mean value for the 2 sample we're
+        comparing, n1 and n2 are the number of people from the 1st ans 2nd
+        sample who provided a response to the survey, Sx1x2 is the standard
+        deviation. In this case the standard deviation is:
+        Sx1x2 = sqrt(((n1-1)*s2x1 + (n2-2)*s2x2)/(n1+n2+2)), where s2x1 and
+        s2x2 are the is the standard deviation for sample 1 and 2.
+
+        :return: numpy 1D array of tstats
+        """
+
+        variance = self._slice.var_scale_means_row
+        # Sum for each column of the counts that have not a nan index in the
+        # related numeric counts
+        not_a_nan_index = ~np.isnan(self._slice.rows_dimension_numeric)
+        counts = np.sum(self._slice.counts[not_a_nan_index, :], axis=0)
+
+        standard_deviation = np.sqrt(
+            np.divide(
+                ((counts[self._col_idx] - 1) * variance[self._col_idx])
+                + ((counts - 1) * np.array(variance)),
+                (counts[self._col_idx] + counts - 2),
+            )
+        )
+
+        tstats_scale_means = (
+            self._slice.scale_means_row[self._col_idx] - self._slice.scale_means_row
+        ) / (standard_deviation * np.sqrt((1 / counts[self._col_idx]) + (1 / counts)))
+
+        return tstats_scale_means
 
     @lazyproperty
     def _df(self):
+        # if the cube to which the slice belongs is a CATxMRxITSELF
```

Comment:
If it's possible i would add a docstring for degree of freedom just summarising the existing comment here